File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[3];
        answer[2] = 1000001;
        int s = 0;
        int e = 0;
        int tmp = sequence[0];
        while (s <= e && e < sequence.length) {
            if (tmp == k) {
                if (e-s+1 < answer[2]) {
                    answer[0] = s;
                    answer[1] = e;
                    answer[2] = e-s+1;
                }
                tmp -= sequence[s];
                s++;
            } else if (tmp > k) {
                tmp -= sequence[s];
                s++;
            } else {
                e++;
                if (e < sequence.length) {
                    tmp += sequence[e];
                }
            }
        }
        int[] result = new int[] {answer[0], answer[1]};
        return result;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    s, e = 0, 0
    tmp = sequence[0]
    while s <= e and e < len(sequence):
        # print(s, e, tmp)
        if tmp == k:
            answer.append((s, e, e-s+1))
            tmp -= sequence[s]
            s += 1
            
        elif tmp > k:
            tmp -= sequence[s]
            s += 1
            
        else:
            e += 1
            if e < len(sequence):
                tmp += sequence[e]
            
    # print('-------------end------')
    # print(s, e, tmp)
    # print(answer)
    answer.sort(key = lambda x : (x[2], x[0]))
    return [answer[0][0], answer[0][1]]